src/main/java/edu/cornell/kfs/coa/document/CUAccountMaintainableImpl.java

```diff
@@ -68,11 +81,27 @@ public void saveBusinessObject() {
         if (isClosingAccount) {
             SpringContext.getBean(AccountReversionTrickleDownInactivationService.class).trickleDownInactivateAccountReversions((Account) getBusinessObject(), getDocumentNumber());
         }
+        
+		// KFSPTS-3877 save xml again so that object id is persisted, this way
+		// notes on the doc will be retrieved form the DB instead of xml doc
+		// content and will display any future notes on account edits as well
+        try {
+            Document document = SpringContext.getBean(DocumentService.class).getByDocumentHeaderId(getDocumentNumber());
+            Account xmlaccount = (Account) ((MaintenanceDocument) document).getNewMaintainableObject().getBusinessObject();
+            if (ObjectUtils.isNull(xmlaccount.getObjectId()) &&( KFSConstants.MAINTENANCE_NEW_ACTION.equals(getMaintenanceAction()) || KFSConstants.MAINTENANCE_COPY_ACTION.equals(getMaintenanceAction()))) {
+                ((MaintenanceDocument) document).getNewMaintainableObject().setBusinessObject(account);
+                SpringContext.getBean(DocumentService.class).saveDocument(document);
```

Comment:
I think the maintenance document will also be saved after performing the doRouteStatusChange() logic that calls the saveBusinessObject() method (but uses a saving method that does not update the maintainable XML). Have you verified that this extra save does not cause the calling code to run into an optimistic locking exception when it tries to do its own document persistence?